website/content/en/preview/testing-guide

```diff
@@ -0,0 +1,41 @@
+# Karpenter Testing User Guide
+
+Currently, users can only test Karpenter by adding to the list of integration tests run in a mock environment or installing and testing Karpenter on a real cluster. Users that want to test more are limited by the lack of automation and configurability, resulting in much less testing.
+
+To increase testing of Karpenter, we will introduce infrastructure and tools to extend the current mechanisms and enable developers to contribute with more confidence.
+
+## Mediums of Testing
+
+__Code Contributions__ are currently tested with integration tests run by GitHub Actions. Testing a contribution comprehensively is hard for a few reasons:
+
+* Testing all supported Kubernetes and Karpenter versions would be very tedious for one developer
+* Testing with real instances would incur unwanted costs to the contributor
+
+__On-demand testing__ is where a developer may need to test Karpenter independent of contribution. Testing here may only rely on some test configurations but could be for the following:
+
+* Scale tests to verify test workloads or dependencies
+* Benchmark performance of custom builds and changes
+
+__Periodic tests__ are meant to run on a pre-determined basis to monitor the health of the Karpenter project. These can be used to:
+
+* Track performance regressions/progressions over time
+* Validate that Karpenter functionally works for existing and future releases.
+
+## User Workflow
+
+__(To be implemented)__ When contributing, users will need to go through steps to automatically test their changes.
+
+* When a user cuts a PR and is ready to test, a maintainer will need to approve testing by posting `/ok-to-test`
+* A robot watching the PR will kick off tests configured in the testing folder that will run in a CI Host Cluster monitored by the Host Cluster.
+* A link to the logs and metrics will be posted in the PR, whether it’s prow as used in the [kubernetes/test-infra](https://github.com/kubernetes/test-infra) or an alternative solution.
+* After the tests are successful, the robot will report the results and a maintainer will merge it once approving the code.
+
+__(To be implemented)__ Users can follow instructions to replicate the infrastructure used to run the tests in the same folder as the tests. Contributors will be able to run tests on their own if they want to shorten the reviewer loop. The associated README will instruct users how to run tests as automated for a PR.
```

Comment:
Do we repeat the tests ourselves even the contributors may have run tests using similar setup?

Reply:
We'll need to ensure that a PR can only be merged after an `/ok-to-test` is successful. I included this as a way for developers to unblock themselves and iterate quicker if maintainers aren't quick enough.